cuqi/solver/_solver.py

```diff
@@ -1,12 +1,12 @@
-import numpy as np
-from numpy import linalg as LA
+import cuqi.array as xp
+LA = xp.linalg  # Linear algebra operations
 from scipy.optimize import fmin_l_bfgs_b, least_squares
 import scipy.optimize as opt
```

Comment:
For scipy. an we add this under the xp also? xp.optimize and xp.sparse? Add NotImplementedError if not implemented in e.g. PyTorch.